Lib/typing.py

```diff
@@ -2008,16 +2008,18 @@ def __instancecheck__(cls, instance):
             raise TypeError("Instance and class checks can only be used with"
                             " @runtime_checkable protocols")
 
+        protocol_attrs = _get_protocol_attrs(cls)
+
         if ((not getattr(cls, '_is_protocol', False) or
```

Comment:
We may want to do this check *before* the `_get_protocol_attrs` call.

Reply:
Good point. I'll try making the change tomorrow, and will redo the benchmarks